### DISCUSSION TITLE
Open URL with system default browser

### DISCUSSION THREAD

**Discussion by zhanghelook** [quasarframework/quasar#13512] [CLI - Electron mode] [answered]
Hi, I'm making an electron desktop app with Quasar framework.

I need to open an URL with user's system default browser (e.g. Chrome, Firefox, etc.)
The goal is like when you click a link in Slack, it opens the link in your system browser.

I tried this `openURL(...)`, but it opens another electron window instead of browser.
https://quasar.dev/quasar-utils/other-utils#open-external-url

I also tried this in electron way,
```typescript
import { shell } from "electron";
shell.openExternal("http://www.google.com")
```
but I get these errors
```
Module not found: Error: Can't resolve 'fs'
Module not found: Error: Can't resolve 'path'
```

Could you give me advice for this please?🧐

**Comment by suleimanas** [comment, marked as answer]
1: in electron-main.js
```
mainWindow.webContents.on('new-window', function(e, url) {
  e.preventDefault();
  shell.openExternal(url);
});

```

2: then use **target="_blank"**
`<a href="https://quasar.dev" target="_blank">Quasar</a>`

**Reply by zhanghelook** [reply]
Thanks. Perfect! I ended up using this as new-window event listener is deprecated in Electron 13
```javascript
  mainWindow.webContents.setWindowOpenHandler(({url}) => {
    shell.openExternal(url)
    return { action: 'deny' }
  })
```
